### PR TITLE
perf: perm sync start time

### PR DESCRIPTION
**Pull request by evan-onyx**
## Description

We were previously starting perm syncs from the beginning of time (or the last perm sync success, when feasible). That is safe and correct, but if the user sets an indexing_start time there's no need to fetch permissions for docs before that time (since they aren't indexed anyways). 

This PR makes it so document permission syncing only happens after indexing_start when the user sets it. There should not be any docs indexed before this time, but in the worst case where somehow there are, the only impact is that users won't be able to see those docs.

## How Has This Been Tested?

unit tests

## Additional Options

- [ ] [Optional] Please cherry-pick this PR to the latest release version.
- [x] [Optional] Override Linear Check

---
## Summary by cubic
Start permission syncing at the connector’s indexing_start (if set) to skip pre-indexed documents. This reduces API calls and speeds up permission sync across connectors.

- **Refactors**
  - Pass `indexing_start` from `cc_pair.connector.indexing_start` to `retrieve_all_slim_docs_perm_sync` in generic and Slack doc sync.
  - Slack: accept `start`/`end` and forward to API as `oldest`/`latest` when listing messages.
  - SharePoint: accept `start`/`end`, convert to UTC `datetime`, and filter drive items and site pages.
  - Confluence: build CQL with `start`/`end` and apply to pages and attachments.
  - Celery task: eager-load connector and credential for `indexing_start` access.
  - Tests: add Jira doc sync tests to verify `start` is propagated (or None); update Slack tests to exercise `oldest`.

<sup>Written for commit 2a3deeec2908d57d8b16857108209fb0468155a7. Summary will update on new commits.</sup>

### CODE REVIEW

**Review by Danelegend — APPROVED** [review]

File: backend/onyx/connectors/slack/connector.py [hunk 1]
```diff
@@ -546,6 +548,8 @@ def _get_all_doc_ids(
             client=client,
             channel=channel,
             callback=callback,
+            oldest=str(start) if start else None,  # 0.0 -> None intentionally
```
[the remark this answers is not in the document]
Comment (evan-onyx) [reply backend/onyx/connectors/slack/connector.py:551]: I left a comment explaining that this is intentional
